diff --git a/modules/nodereference/nodereference.module b/modules/nodereference/nodereference.module
index 30b57f5..43bc560 100755
--- a/modules/nodereference/nodereference.module
+++ b/modules/nodereference/nodereference.module
@@ -872,7 +872,12 @@ function _nodereference_potential_references_views($field, $string = '', $match
 
     // Limit result set size.
     $limit = isset($limit) ? $limit : 0;
-    $view->display_handler->set_option('items_per_page', $limit);
+    if (isset($view->api_version) && intval(substr($view->api_version, 0, 1)) >= 3) {
+      $view->set_items_per_page($limit);
+    }
+    else {
+      $view->display_handler->set_option('items_per_page', $limit);
+    }
 
     // Get arguments for the view.
     if (!empty($field['advanced_view_args'])) {
